### PR TITLE
build: Display feature detect result of configure

### PR DESCRIPTION
**Pull request by honggyukim**
It'd be better to display the result of feature detect when running
configure script.
```
  $ ./configure --prefix=$HOME/usr --without-libdw
  uftrace detected system features:
  ...                        prefix: /home/honggyu/usr
  ...                        libelf: [ on  ]
  ...                         libdw: [ OFF ]
  ...                  libpython2.7: [ on  ]
  ...                   libncursesw: [ on  ]
  ...                  cxa_demangle: [ on  ]
```
It shows the outlook almost same as Linux perf tool.

Signed-off-by: Honggyu Kim <honggyu.kp@gmail.com>

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -138,6 +138,28 @@ for dep in $IGNORE; do
...
+
+print_feature()
+{
+    item=$1
+    file=$2
+
+    if [ -f ${srcdir}/check-deps/${file} ]; then
+        onoff="\033[32mon \033[0m"
+    else
+        onoff="\033[91mOFF\033[0m"
```
[the remark this answers is not in the document]
Comment (honggyukim) [reply configure]: Thanks. Just updated after checking it with `if [ -t 1 ]`.

**Comment by namhyung** [comment]
It looks like doing same job as perf does.  Did you copy some code from it?  If so it'd be nice to write it in the changelog.

**Comment by honggyukim** [comment]
Right. I made the outlook same as perf but didn't use any code from perf.

**Comment by honggyukim** [comment]
As you mentioned, I'd be better to write that I got the outlook from Linux perf so updated the commit message including it.

**Comment by honggyukim** [comment]
I think other users might be wondering about the meaning of each field so I'm just wondering how you think about the below output:
```
$ ./configure --prefix=$HOME/usr --without-libdw
uftrace detected system features:
...         prefix: /home/honggyu/usr
...         libelf: [ on  ] - use libelf or use internal raw ELF parser
...          libdw: [ OFF ] - use debug info to detect arguments/return type
...   libpython2.7: [ on  ] - python scripting support
...    libncursesw: [ on  ] - TUI support
...   cxa_demangle: [ on  ] - full demangler support
```

**Comment by honggyukim** [comment]
@namhyung If you agree on it, I will update this PR.
```diff
+diff --git a/configure b/configure
index 520566d..e7bd325 100755
--- a/configure
+++ b/configure
@@ -144,6 +144,7 @@ print_feature()
 {
     item=$1
     file=$2
+    description=$3

     if [ -t 1 ]; then
         # use colored output only when stdout is tty
@@ -159,15 +160,15 @@ print_feature()
             onoff="OFF"
         fi
     fi
-    printf "...%30s: [ ${onoff} ]\n" "${item}"
+    printf "...%15s: [ ${onoff} ] - %s\n" "${item}" "${description}"
 }

-printf "...%30s: %s\n" "prefix" "${prefix}"
-print_feature "libelf" "have_libelf"
-print_feature "libdw" "have_libdw"
-print_feature "libpython2.7" "have_libpython2.7"
-print_feature "libncursesw" "have_libncurses"
-print_feature "cxa_demangle" "cxa_demangle"
+printf "...%15s: %s\n" "prefix" "${prefix}"
+print_feature "libelf" "have_libelf" "use libelf or use internal raw ELF parser"
+print_feature "libdw" "have_libdw" "use debug info to detect arguments/return type"
+print_feature "libpython2.7" "have_libpython2.7" "python scripting support"
+print_feature "libncursesw" "have_libncurses" "TUI support"
+print_feature "cxa_demangle" "cxa_demangle" "full demangler support"

 cat >$output <<EOF
 # this file is generated automatically
```

**Comment by honggyukim** [comment]
I also think that we can provide this info to the output of `uftrace --version`.

**Comment by namhyung** [comment]
Looks good.  I'd like to change it a little bit:
 * libelf - more flexible ELF data handling
 * libdw - DWARF debug info support (for argument, location, ...)

Also the version string is printed from argp runtime so you should make it static const string.

**Comment by honggyukim** [comment]
@namhyung Thanks for your comments. I just updated as you guided.

ChangeLog:
- follow description from @namhyung for libelf and libdw
- add dependency info to version output

File: configure [hunk 2]
```diff
@@ -138,6 +138,38 @@ for dep in $IGNORE; do
...
+        else
+            onoff="OFF"
+        fi
+    fi
+    printf "...%15s: [ ${onoff} ] - %s\n" "${item}" "${description}"
+}
+
+printf "...%15s: %s\n" "prefix" "${prefix}"
+print_feature "libelf" "have_libelf" "more flexible ELF data handling"
+print_feature "libdw" "have_libdw" "DWARF debug info support (for argument, location, ...)"
```
Comment (namhyung) [inline configure]: I'm sorry but it seems bit long so let's removed the parenthesis part.

**Comment by honggyukim** [comment]
The output looks as follows with this change:
```
$ ./configure --prefix=$HOME/usr
uftrace detected system features:
...         prefix: /home/honggyu/usr
...         libelf: [ on  ] - more flexible ELF data handling
...          libdw: [ on  ] - DWARF debug info support (for argument, location, ...)
...   libpython2.7: [ on  ] - python scripting support
...    libncursesw: [ on  ] - TUI support
...   cxa_demangle: [ on  ] - full demangler support using libstdc++

$ make install
    ...
$ uftrace --version
uftrace v0.8.3-344-g3cfc ( dwarf python tui )
```

File: misc/version.sh [hunk 3]
```diff
@@ -26,9 +28,23 @@ if test -z "${GIT_VERSION}" -a -n "${FILE_VERSION}"; then
...
+    DEPS="${DEPS} tui"
+fi
+if [ "x${DEPS}" != "x" ]; then
+    DEPS="(${DEPS} )"
+fi
+
 if test -z "${FILE_VERSION}" -o "${CURR_VERSION}" != "${FILE_VERSION}"; then
     # update file version only if it's different
-    echo "#define UFTRACE_VERSION  \"${CURR_VERSION}\"" > ${VERSION_FILE}
+    echo "#define UFTRACE_VERSION  \"${CURR_VERSION} ${DEPS}\"" > ${VERSION_FILE}
```
Comment (namhyung) [inline misc/version.sh]: Nice work!  This way the output of uftrace info will also be changed.

I have a little suggestion.  It'd be nice to have no space if no dependency is detected.  So how about this?

```
if [ "x${DEPS} != "x" ]; then
  DEPS=" (${DEPS} )"
fi
...
  echo "#define UFTRACE_VERSION  \"${CURR_VERSION}${DEPS}\"" > ${VERSION_FILE}
```

File: configure (hunk 2 above)
Comment (honggyukim) [reply configure]: No problem.

File: misc/version.sh (hunk 3 above)
Comment (honggyukim) [reply misc/version.sh]: Sure.

**Comment by honggyukim** [comment]
I just updated it again. Thanks for the review!

**Comment by honggyukim** [comment]
Here is the (maybe) final output.
```
$ ./configure --prefix=$HOME/usr
uftrace detected system features:
...         prefix: /home/honggyu/usr
...         libelf: [ on  ] - more flexible ELF data handling
...          libdw: [ on  ] - DWARF debug info support
...   libpython2.7: [ on  ] - python scripting support
...    libncursesw: [ on  ] - TUI support
...   cxa_demangle: [ on  ] - full demangler support with libstdc++

$ make
    ...
$ uftrace --version
uftrace v0.8.3-344-gb8eb8 ( dwarf python tui )
```

**Pull request merged**